src/core/shared_memory_manager.cc

```diff
@@ -205,6 +211,23 @@ SharedMemoryManager::UnregisterAllSharedMemory()
   return Status::Success;
```

Comment:
Above you shouldn't just log an error you should also return the same message as an error status.